### DISCUSSION TITLE
Support for Athena Iceberg tables

### DISCUSSION THREAD

**Discussion by andyadamides** [aws/aws-sdk-pandas#1399] [General]
Hi, based on the latest version of AWS Data Wrangler, I cannot see any support for Athena [Iceberg](https://docs.aws.amazon.com/athena/latest/ug/querying-iceberg-creating-tables.html) tables. Is this on the roadmap?

**Comment by jaidisido** [comment]
I am not sure what kind of support you have in mind? Existing wrangler Athena methods should work on Iceberg tables the same way they do on other types

**Reply by andyadamides** [reply]
I have played with existing functions for 30min or so, and I wasn't able to get it to work. I have tried `s3.to_parquet` method and added `table_type` as `ICEBERG`. The write works from AWS Data Wrangler, i.e. no exception. But when I get to `Glue` I don't see the `ICEBERG` tag in the table details which is normally visible if you create an SQL table from Athena with
```sql
CREATE TABLE table_iceberg_4 (col1 string, col2 string, col3 date)
PARTITIONED BY (col3)  
  LOCATION 's3://location/location'
  TBLPROPERTIES ( 'table_type' = 'ICEBERG' )
```

Also, when I try a `DELETE FROM _ WHERE _` Athena SQL query(which is what Iceberg is supposed to be for) in that new table it gives me 
```
NOT_SUPPORTED: This connector only supports delete where one or more partitions are deleted entirely
```

So before spending any more type on this, thought to ask here if you had any ideas or any examples on how this works with AWS Data Wrangler?

**Reply by jaidisido** [reply]
Ah I see, so you are not referring to the `wr.athena.*` methods per se (i.e. the ability to query/update/delete from an Iceberg Athena table) but more to the ability to create an Iceberg table in Glue when writing data to S3 for example. That is not covered at the moment indeed, but I cannot see why not. I will transfer to a new issue and the team can look into it, thanks!

**Reply by andyadamides** [reply]
Yea well `s3.to_parquet` would create an athena/glue table if there wasn't one present, so that is why I mentioned it like the way I did. Can you share the issue, so that I can track the progress? Cheers
